### PR TITLE
bugfix: updated the value of remaining when `count` changes in limit-count plugin.

### PR DESCRIPTION
**Pull request by gy09535**
### What this PR does / why we need it:
fix https://github.com/apache/apisix/issues/2098

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by moonming** [comment]
@gy09535 please add test cases, thx

**Comment by gy09535** [comment]
> @gy09535 please add test cases, thx
added

File: apisix/init.lua [hunk 1]
```diff
@@ -361,21 +361,12 @@ function _M.http_access_phase()
...
-                              .. service.value.id
-        else
-            api_ctx.conf_type = "service"
-            api_ctx.conf_version = service.modifiedIndex
-            api_ctx.conf_id = service.value.id
-        end
+        api_ctx.conf_type = "route&service"
+        api_ctx.conf_version = route.modifiedIndex .. "&" .. service.modifiedIndex
+        api_ctx.conf_id = route.value.id .. "&"
+                .. service.value.id
```
Comment (membphis) [inline apisix/init.lua:369]: bad indentation

File: apisix/plugin.lua [hunk 2]
```diff
@@ -297,6 +297,7 @@ local function merge_service_route(service_conf, route_conf)
     local new_conf = core.table.deepcopy(service_conf)
     new_conf.value.service_id = new_conf.value.id
     new_conf.value.id = route_conf.value.id
+    new_conf.modifiedIndex=route_conf.modifiedIndex
```
Comment (membphis) [inline apisix/plugin.lua:300]: `new_conf.modifiedIndex=route_conf.modifiedIndex`
->
`new_conf.modifiedIndex = route_conf.modifiedIndex`

File: apisix/plugin.lua [hunk 3]
```diff
@@ -332,7 +333,8 @@ function _M.merge_service_route(service_conf, route_conf)
     core.log.info("service conf: ", core.json.delay_encode(service_conf))
     core.log.info("  route conf: ", core.json.delay_encode(route_conf))
 
-    return merged_route(route_conf, service_conf,
+    local rout_service_key = route_conf.modifiedIndex .. service_conf.modifiedIndex
```
Comment (membphis) [inline apisix/plugin.lua:336]: add a separator: `modifiedIndex .. "#" .. modifiedIndex`

File: t/plugin/limit-count.t [hunk 4]
```diff
@@ -637,3 +637,54 @@ GET /t
 passed
 --- no_error_log
 [error]
+
```
Comment (membphis) [inline t/plugin/limit-count.t:640]: three blank lines between different test cases

File: t/plugin/limit-count.t [hunk 5]
```diff
@@ -637,3 +637,54 @@ GET /t
...
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }   
+
```
Comment (membphis) [inline t/plugin/limit-count.t:684]: remove this line

**Review by membphis — CHANGES_REQUESTED** [review]

**Comment by gy09535** [comment]
thanks  for reviewing , all required format has fixed.

**Review by membphis — APPROVED** [review]

File: apisix/init.lua [hunk 6]
```diff
@@ -361,21 +361,11 @@ function _M.http_access_phase()
             return core.response.exit(404)
         end
 
-        local changed
-        route, changed = plugin.merge_service_route(service, route)
+        route= plugin.merge_service_route(service, route)
```
Comment (moonming) [inline apisix/init.lua:364]: need a space before `=`

File: apisix/plugin.lua [hunk 7]
```diff
@@ -332,7 +333,8 @@ function _M.merge_service_route(service_conf, route_conf)
     core.log.info("service conf: ", core.json.delay_encode(service_conf))
     core.log.info("  route conf: ", core.json.delay_encode(route_conf))
 
-    return merged_route(route_conf, service_conf,
+    local rout_service_key = route_conf.modifiedIndex .. "#" .. service_conf.modifiedIndex
```
Comment (moonming) [inline apisix/plugin.lua:336]: `rout_service_key` -> `route_service_key`

File: apisix/plugin.lua [hunk 8]
```diff
@@ -297,6 +297,7 @@ local function merge_service_route(service_conf, route_conf)
     local new_conf = core.table.deepcopy(service_conf)
     new_conf.value.service_id = new_conf.value.id
     new_conf.value.id = route_conf.value.id
+    new_conf.modifiedIndex = route_conf.modifiedIndex
```
Comment (moonming) [inline apisix/plugin.lua:300]: `new_conf.modifiedIndex` -> new_conf.modified_index

Comment (gy09535) [reply apisix/plugin.lua:300]: the "modifiedIndex" while be used in other space as route modified Index, if I change this key ,should change in all place.

File: t/plugin/limit-count.t [hunk 9]
```diff
@@ -637,3 +637,55 @@ GET /t
 passed
 --- no_error_log
 [error]
+
+
+
+=== TEST 21: when the count is changed, check  the limit is correct
```
Comment (moonming) [inline t/plugin/limit-count.t:643]: need add test case to cover: `count` from 1 to 2.

Comment (moonming) [reply t/plugin/limit-count.t:643]: and `count` from 2 to 2.

File: apisix/plugin.lua (hunk 8 above)
Comment (moonming) [reply apisix/plugin.lua:300]: got it

**Review by moonming — CHANGES_REQUESTED** [review]

File: t/plugin/limit-count.t (hunk 9 above)
Comment (gy09535) [reply t/plugin/limit-count.t:643]: added

**Review by moonming — APPROVED** [review]

**Comment by moonming** [comment]
```
#   Failed test 'TEST 1: log rotate - response_body_like - response is expected (2020-08-31_10-25-32__access.log 2020-08-31_10-25-31__access.log 2020-08-31_10-25-33__access.log . 2020-08-31_10-25-32__error.log error.log 2020-08-31_10-25-33__error.log access.log 2020-08-31_10-25-31__error.log .. nginx.pid)'
#   at /home/runner/work/apisix/apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1635.
#                   '2020-08-31_10-25-32__access.log
# 2020-08-31_10-25-31__access.log
# 2020-08-31_10-25-33__access.log
# .
# 2020-08-31_10-25-32__error.log
# error.log
# 2020-08-31_10-25-33__error.log
# access.log
# 2020-08-31_10-25-31__error.log
# ..
# nginx.pid
# '
#     doesn't match '(?^:\_error\.log[\s\S]*\_access\.log)'
# Looks like you failed 1 test of 7.
t/plugin/log-rotate.t .................... 
```
@membphis please take a look this failed test case.

**Comment by membphis** [comment]
> #   at /home/runner/work/apisix/apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1635.
> #                   '2020-08-31_10-25-32__access.log
> # 2020-08-31_10-25-31__access.log
> # 2020-08-31_10-25-33__access.log
> # .
> # 2020-08-31_10-25-32__error.log
> # error.log
> # 2020-08-31_10-25-33__error.log
> # access.log
> # 2020-08-31_10-25-31__error.log
> # ..
> # nginx.pid
> # '
> #     doesn't match '(?^:\_error\.log[\s\S]*\_access\.log)'
> # Looks like you failed 1 test of 7.
> t/plugin/log-rotate.t .................... 
> ```

we need to update this test case.  will create a new issue later

**Comment by gy09535** [comment]
Is their anything I can do for this PR?

**Comment by membphis** [comment]
it the test case is unstable, we need to merge https://github.com/apache/apisix/pull/2164 first

**Pull request merged**